src/tools/haxelib/Main.hx

```diff
@@ -951,39 +953,35 @@ class Main {
 			print("Checking " + p);
 			try doUpdate(p, state)
 			catch (e:Dynamic)
-				if (e != GIT_UNAVAILABLE) neko.Lib.rethrow(e);
+				if (e != VCS_UNAVAILABLE) neko.Lib.rethrow(e);
 		}
 		if( state.updated )
 			print("Done");
 		else
 			print("All libraries are up-to-date");
 	}
-	
-	static var GIT_UNAVAILABLE = 'Git unavailable';
+
+	static var VCS_UNAVAILABLE = 'Git unavailable';
 	function doUpdate( p : String, state ) {
 		var rep = state.rep;
-		if( FileSystem.exists(rep + "/" + p + "/git") && FileSystem.isDirectory(rep + "/" + p + "/git") ) {
-			checkGit();
-			if (!gitExists())
-				throw GIT_UNAVAILABLE;
+
+		//TODO: get content from `.current` and use vcs only if there "dev".
+
+		var vcs:Vcs = Vcs.getVcsForDevLib(rep + "/" + p);
+		if(vcs != null)
+		{
+			if(!vcs.exists)
+				throw VCS_UNAVAILABLE;
```

Comment:
Given that this is a constant, the displayed error will not be helpful. I think it would be better to throw the `vcs` (allowing for a `catch (e:Vcs) {}`) or have an explicit error for that.